sale_purchase_customer/views/purchase_order.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="purchase_order_form_view" model="ir.ui.view">
+        <field name="name">purchase.order.form.view</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='partner_ref']" position="after">
+                <field name="sale_partner_id" attrs="{'readonly':[('state','in',['done','cancel'])]}"/>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="purchase_order_search_view" model="ir.ui.view">
```

Comment:
In this case, Odoo has splitted search views for RFQs and Purchase Orders. And this one is only for RFQs, so please add the same behavior to Purchase Orders search